src/intervals/real_interface.jl

```diff
@@ -78,96 +78,77 @@ Base.hash(x::Interval, h::UInt) = hash(sup(x), hash(inf(x), hash(Interval, h)))
+Base.union!(::AbstractVector{S}, ::Interval, ::Any, ::Any...) where {S} =
+    throw(ArgumentError("`union!` is purposely not supported for intervals. See instead `hull`"))
+Base.union!(::AbstractVector{S}, ::Interval, ::Interval, ::Any...) where {S} =
+    throw(ArgumentError("`union!` is purposely not supported for intervals. See instead `hull`"))
+Base.union!(::AbstractVector{S}, ::Any, ::Interval, ::Any...) where {S} =
     throw(ArgumentError("`union!` is purposely not supported for intervals. See instead `hull`"))
 
+Base.setdiff(::Interval) =
+    throw(ArgumentError("`setdiff` is purposely not supported for intervals. See instead `interiordiff`"))
+Base.setdiff!(::AbstractSet, ::Interval) =
+    throw(ArgumentError("`setdiff!` is purposely not supported for intervals. See instead `interiordiff`"))
 
 # pointwise equality
 
-"""
-    ==(::BareInterval, ::Number)
-    ==(::Number, ::BareInterval)
-    ==(::Interval, ::Number)
-    ==(::Number, ::Interval)
-
-Test whether an interval is the singleton of a given number. In other words, the
-result is true if and only if the interval contains only that number.
-
-!!! note
-    Comparison between intervals is purposely disallowed. Indeed, equality
-    between non-singleton intervals has distinct properties, notably ``x = y``
-    does not imply ``x - y = 0``. See instead [`isequal_interval`](@ref).
-"""
-Base.:(==)(x::Union{BareInterval,Interval}, y::Number) = isthin(x, y)
-Base.:(==)(x::Number, y::Union{BareInterval,Interval}) = y == x
+Base.:(==)(x::Interval, y::Number) = !isthin(x) & in_interval(y, x) ? throw(ArgumentError("`==` is purposely not supported when the number is contained in the interval. See instead `isthin`")) : isthin(x, y)
+Base.:(==)(x::Number, y::Interval) = y == x
 # needed to resolve ambiguity from irrationals.jl
-Base.:(==)(x::Interval, y::AbstractIrrational) = isthin(x, y)
+Base.:(==)(x::Interval, y::AbstractIrrational) = !isthin(x) & in_interval(y, x) ? throw(ArgumentError("`==` is purposely not supported when the number is contained in the interval. See instead `isthin`")) : isthin(x, y)
 Base.:(==)(x::AbstractIrrational, y::Interval) = y == x
 # needed to resolve ambiguity from complex.jl
 Base.:(==)(x::Interval, y::Complex) = isreal(y) & (real(y) == x)
 Base.:(==)(x::Complex, y::Interval) = y == x
 
-# follows docstring of `Base.iszero`
-Base.iszero(x::Union{BareInterval,Interval}) = isthinzero(x)
 
-# follows docstring of `Base.isone`
-Base.isone(x::Union{BareInterval,Interval}) = isthinone(x)
+Base.:<(x::Interval, y::Real) = (!isthin(x) & in_interval(y, x)) | isempty_interval(x) ? throw(ArgumentError("`<` is purposely not supported when the number is contained in the interval, or if the interval is empty")) : sup(x) < y
+Base.:<(x::Real, y::Interval) = (!isthin(y) & in_interval(x, y)) | isempty_interval(y) ? throw(ArgumentError("`<` is purposely not supported when the number is contained in the interval, or if the interval is empty")) : x < inf(y)
```

Comment:
```suggestion
Base.:<(x::Interval, y::Real) = ((!isthin(x) & in_interval(y, x)) | isempty_interval(x)) ? throw(ArgumentError("`<` is purposely not supported when the number is contained in the interval, or if the interval is empty")) : sup(x) < y
Base.:<(x::Real, y::Interval) = ((!isthin(y) & in_interval(x, y)) | isempty_interval(y)) ? throw(ArgumentError("`<` is purposely not supported when the number is contained in the interval, or if the interval is empty")) : x < inf(y)
```

I got tripped by the precedence of `?` compared to `|`.

Why are `|` and `&` used instead of `||` and `&&` here ?